Makefile

```diff
@@ -128,16 +129,16 @@ devtools-stamp: tools
 devtools-clean: tools-clean
 	rm -f devtools-stamp
 
-vendor-deps: tools
-	@echo "--> Generating vendor directory via dep ensure"
-	@rm -rf .vendor-new
-	@dep ensure -v -vendor-only
-	tar -c vendor/ | sha1sum | cut -d' ' -f1 > $@
+go-mod-cache: go.sum
+	@echo "--> Download go modules to local cache"
+	@go mod download
```

Comment:
Do we need a separate `go mod download` command? Won't any call that needs the dependencies trigger this (e.g. go build / go test etc). Is this some trick for caching or do you otherwise need a separate step in the Makefile?

Reply:
I tried to make it w/o calling `download`. I noticed that go was caching things multiple times

Reply:
> go was caching things multiple times

What do you mean by that? How did you notice?

Reply:
By looking at the CI steps. Each of them were caching packages.

Reply:
OK, I think I understand what is going on: For the simulations several `go build` calls are run in parallel, or at least one `go test` call is kicked off before the first finished? 

The scripts look like there is a sleep in between calls (10s):
https://github.com/cosmos/cosmos-sdk/blob/3d2886dc496be82a66576908818c85284e96e93f/scripts/multisim.sh#L33-L37

This is related: https://github.com/golang/go/issues/26794

I think it could be more reasonable to warm up the cache in those scripts where it is needed. Meaning explicitly calling `go mod download` directly in those simulation scripts (and not adding it to the Makefile). That way we would not need to handle this differently across our repos. What do you think?

Reply:
Yeah, I'm ok with that